### DISCUSSION TITLE
How to set index sequence for different types of nodes in the to_homogeneous() function

### DISCUSSION THREAD

**Discussion by TheAllUnknown** [pyg-team/pytorch_geometric#8733] [Q&A] [answered]
I have the Heterodata object from MovieLens, there are two types of nodes: movie and user, with only one edge type : `('user','rates','movie')`, and I want to add a reverse edge `('movie','rated_by','user')` and then transform the graph to homogeneous. This is how I did the work:
```
edge_index = data[('user','rates','movie')]['edge_index'][[1,0],:]
data[('movie','rated_by','user')]['edge_index'] = edge_index
homodata = data.to_homogeneous()
```

everything is good except the index, I want the index for the user node to start from 0, and the index for the movie node to start from `num_users`, while the result gives me the contrary version. How should I fix this?

**Comment by rusty1s** [comment]
Currently, the order is defined by the ordering of node types in the `HeteroData.node_store_dict`.  The best way to enforce a user-defined ordering would be to re-order the dictionary:
```python
data._node_store_dict = {key: data._node_store_dict[key] for key in ["user", "movie"]}

**Reply by TheAllUnknown** [reply]
Thanks, but it looks like when giving value to `_node_store_dict`, this attribute remains unchanged when called again.
```
data._node_store_dict = {key: data._node_store_dict[key] for key in ["user", "movie"]}
print(data._node_store_dict)
```
and I get result still the same:
```
{'movie': {'x': tensor([[-0.0828,  0.0530,  0.0536,  ...,  0.0000,  0.0000,  0.0000],
        [-0.1053,  0.1508, -0.0264,  ...,  0.0000,  0.0000,  0.0000],
        [-0.0988,  0.0176, -0.0527,  ...,  0.0000,  0.0000,  0.0000],
        ...,
        [-0.1115,  0.0310, -0.0177,  ...,  0.0000,  0.0000,  0.0000],
        [ 0.0366,  0.0137,  0.0315,  ...,  0.0000,  0.0000,  0.0000],
        [-0.0500, -0.0141, -0.0031,  ...,  0.0000,  0.0000,  0.0000]])}, 'user': {'num_nodes': 610}}
```

**Reply by rusty1s** [reply, marked as answer]
Mh, yeah, sorry. It needs to be
```python
data.__dict__['_node_store_dict'] = {
    key: data._node_store_dict[key]
    for key in ["user", "movie"]
}
```
I think we could add a small helper function for this to make this usage easier. Feel free to submit a PR if you have time :)

**Reply by TheAllUnknown** [reply]
Thanks, I can give it a try.
